parsely/src/main/java/com/parsely/parselyandroid/ParselyTracker.kt

```diff
@@ -24,76 +25,95 @@ import android.content.Context
         videoMetadata: ParselyVideoMetadata,
         extraData: Map<String, Any>? = null,
-    )
-    public fun trackPause()
-    public fun resetVideo()
-    public fun flushEventQueue()
-    public fun flushTimerIsActive(): Boolean
-
-    public companion object {
-        private const val DEFAULT_FLUSH_INTERVAL_SECS = 60
-        private var instance: ParselyTrackerInternal? = null
-
-        /**
-         * Singleton instance accessor. Note: This must be called after [.sharedInstance]
-         *
-         * @return The singleton instance
-         */
-        @JvmStatic
-        public fun sharedInstance(): ParselyTracker? {
-            return instance
-        }
+    ): Unit = ensureInitialized().trackPlay(url, urlRef, videoMetadata, extraData)
 
-        /**
-         * Singleton instance factory Note: this must be called before [.sharedInstance]
-         *
-         * @param siteId        The Parsely public site id (eg "example.com")
-         * @param flushInterval The interval at which the events queue should flush, in seconds
-         * @param context             The current Android application context
-         * @param dryRun If set to `true`, events **won't** be sent to Parse.ly server
-         * @return The singleton instance
-         */
-        @JvmStatic
-        @JvmOverloads
-        public fun sharedInstance(
-            siteId: String,
-            flushInterval: Int = DEFAULT_FLUSH_INTERVAL_SECS,
-            context: Context,
-            dryRun: Boolean = false,
-        ): ParselyTracker {
-            return instance ?: run {
-                val newInstance = ParselyTrackerInternal(siteId, flushInterval, context, dryRun)
-                instance = newInstance
-                return newInstance
-            }
+    @JvmStatic
+    public fun trackPause(): Unit = ensureInitialized().trackPause()
+
+    @JvmStatic
+    public fun resetVideo(): Unit = ensureInitialized().resetVideo()
+
+    @JvmStatic
+    public fun flushEventQueue(): Unit = ensureInitialized().flushEventQueue()
+
+    @JvmStatic
+    public fun flushTimerIsActive(): Boolean = ensureInitialized().flushTimerIsActive()
+
+    /**
+     * Singleton instance factory Note: this must be called before [.sharedInstance]
+     *
+     * @param siteId        The Parsely public site id (eg "example.com")
+     * @param flushInterval The interval at which the events queue should flush, in seconds
+     * @param context             The current Android application context
+     * @param dryRun If set to `true`, events **won't** be sent to Parse.ly server
+     * @return The singleton instance
+     */
+    @JvmStatic
+    @JvmOverloads
+    public fun init(
+        siteId: String,
+        flushInterval: Int = DEFAULT_FLUSH_INTERVAL_SECS,
+        context: Context,
+        dryRun: Boolean = false,
+    ) {
+        if (instance != null) {
+            Logging.log("Parse.ly SDK has been already initialized. Reinitialization is not supported.")
```

Comment:
Now that this is in an `init` method, I think an exception is more appropriate than a log.

Reply:
I've introduced it in 1ddfda1 and tested in 0478d9e.

I'm struggling with my thoughts though - I understand the reasoning of signaling to the API consumers that they're doing something that they shouldn't as fast as possible. On the other hand, I'm not entirely sure if the analytics SDK should crash the app in case of such "not destructive" event.

Nevertheless, I'm not convinced strongly to any side, so I'm aligning to your comment. Just wanted to share this thought.

Reply:
That's a good point - I treated this as if it was something internal. As a consumer, I like it crashing instead of silent failures, but I could see other devs preferring a non-destructive approach to this issue. So, I am not sure which is better either.

I'll leave the final decision to you.

Reply:
I think we can leave the exception for now and act on it if we'll have feedback about this from SDK consumers. It'll be easier to update SDK from "fail fast" to "do not fail" rather than other way around.